src/AlarmClock.h

```diff
@@ -0,0 +1,140 @@
+/* 
+ * File:   AlarmClock.h
+ * Author: Craig Cogdill
+ * Created: October 15, 2015 10:45am
+ */
+
+#pragma once
+#include <chrono>
+#include <thread>
+#include <atomic>
+#include <future>
+#include "StopWatch.h"
+
+
+template<typename Duration> class AlarmClock {
+public:
+   typedef std::chrono::microseconds microseconds;
+   typedef std::chrono::milliseconds milliseconds;
+   typedef std::chrono::seconds seconds;
+   
+   AlarmClock(unsigned int sleepDuration) : mExpired(false),
+      kSleepTime(sleepDuration),
+      kSleepTimeMs(ConvertToMilliseconds(Duration(kSleepTime))),
+      kSleepTimeUs(ConvertToMicroseconds(Duration(kSleepTime))),
+      mKeepRunning(new bool(true)),
+      mExited(std::async(std::launch::async,
+                         &AlarmClock::AlarmClockThread,
+                         this,
+                         mKeepRunning)) {
+   }
+   
+   virtual ~AlarmClock() {
+      mKeepRunning = false;
```

Comment:
Please add a test for this to run 100 times,. (or something) each time AlarmClock should be set to run for 1 minute and almost immediately after you delete the object. 

The AlarmClock destruction should then be almost instantaneous.   If the raced induced by using shared_ptr and not shared_ptr<atomic<... is an issue then hopefully this test will cover that.